32/berubejd/selenium_testing.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env python3.8
+
+from pathlib import Path
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from time import sleep
+
+
+def setup(url=None):
+    if not url:
+        url = "http://pyplanet.herokuapp.com/"
+
+    print("\tStarting browser setup...")
+
+    chrome_driver = Path.cwd() / Path("chromedriver.exe")
+    driver = webdriver.Chrome(executable_path=chrome_driver)
+    driver.get(url)
+
+    return driver
+
+
+def test_homepage(driver):
+    """ Test homepage header and nav links """
+
+    print("\tPerforming anonymous user homepage testing...")
+
+    # Header
+    header = driver.find_element_by_xpath("/html/body/header/h1")
+    assert (
+        "PyBites 100 Days of Django" in header.text
+    ), "Incorrect header text or text missing"
+
+    # Login link
+    login_link = driver.find_element_by_xpath('//*[@id="login"]/a[1]')
+    assert "Login" in login_link.text, "Incorrect login link or link missing"
+
+    # Home link
+    home_link = driver.find_element_by_xpath('//*[@id="login"]/a[2]')
+    assert "Home" in home_link.text, "Incorrect home link or link missing"
+
+
+def test_sharer_app(driver):
+    """ Test sharer app link and generated table """
+
+    print("\tPerforming anonymous user App testing...")
+
+    # Navigate to the Sharer App
+    driver.find_element_by_xpath("/html/body/main/ul/li/a").click()
+
+    # Collect table header and links
+    table_header = driver.find_element_by_xpath("/html/body/main/table/thead")
+    assert "Title" in table_header.text, "Incorrect table header"
+
+    table_links = driver.find_elements_by_xpath("/html/body/main/table/tbody/tr/td/a")
+    assert len(table_links) == 100, "Incorrect number of table rows"
```

Comment:
You might wish to make 1000 into a CONSTANT at the top so it's easily updated should this value need to change.

Reply:
Agreed.  This is a better pattern.

Reply:
Moved to a constant.